### PR TITLE
SDN-4916: Bump to k8s v1.30 latest

### PR DESCRIPTION
**Pull request by jcaamano**
Refer to commit messages for detailed steps taken.

Replaces: #2423

### CODE REVIEW

File: pkg/controller/clusterconfig/clusterconfig_controller.go [hunk 1]
```diff
@@ -43,7 +44,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	// Watch for changes to primary resource config.openshift.io/v1/Network
-	err = c.Watch(source.Kind(mgr.GetCache(), &configv1.Network{}), &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{})
+	err = c.Watch(source.Kind[client.Object](mgr.GetCache(), &configv1.Network{}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}))
```
Comment (kyrtapz) [inline pkg/controller/clusterconfig/clusterconfig_controller.go:47]: For consistency with other files can we use a `crclient` alias for `sigs.k8s.io/controller-runtime/pkg/client`?

**Comment by jcaamano** [comment]
/retitle SDN-4923: Bump to k8s v1.30 latest

**Pull request merged**